### PR TITLE
port set-process-sentinel

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/process.rs [hunk 1]
```diff
@@ -287,6 +287,31 @@ pub fn set_process_buffer(process: LispObject, buffer: LispObject) -> LispObject
...
+pub fn set_process_sentinel(process: LispObject, mut sentinel: LispObject) -> LispObject {
+    let mut p_ref = process.as_process_or_error();
+    if sentinel.is_nil() {
+        sentinel = Qinternal_default_process_sentinel;
+    }
+    unsafe { pset_sentinel(p_ref.as_mut(), sentinel.to_raw()) }
+    let process_type = LispObject::from_raw(p_ref.process_type);
+    if process_type.eq(LispObject::from_raw(Qnetwork))
+        || process_type.eq(LispObject::from_raw(Qserial))
+        || process_type.eq(LispObject::from_raw(Qpipe))
```
Comment (shaleh) [inline rust_src/src/process.rs]: Is there a reason we are not using symbolic names like the C code?

File: rust_src/src/process.rs [hunk 2]
```diff
@@ -287,6 +287,31 @@ pub fn set_process_buffer(process: LispObject, buffer: LispObject) -> LispObject
...
+/// Give PROCESS the sentinel SENTINEL; nil for default.
+/// The sentinel is called as a function when the process changes state.
+/// It gets two arguments: the process, and a string describing the change.
+#[lisp_fn]
+pub fn set_process_sentinel(process: LispObject, mut sentinel: LispObject) -> LispObject {
+    let mut p_ref = process.as_process_or_error();
+    if sentinel.is_nil() {
+        sentinel = Qinternal_default_process_sentinel;
+    }
+    unsafe { pset_sentinel(p_ref.as_mut(), sentinel.to_raw()) }
```
[the remark this answers is not in the document]
Comment (brotzeit) [reply rust_src/src/process.rs]: I never really understood why we had LispObject and Lisp_Object. What was the idea behind that ?

[the remark this answers is not in the document]
Comment (shaleh) [reply rust_src/src/process.rs]: We needed it for a while to catch mistakes in porting between C and back.

**Comment by shaleh** [comment]
Thanks @Wilfred! :+1:

**Pull request merged**